@pwinckles Thank you for such a quick turnaround!  Happy to test and I did.  Unfortunately, I'm receiving errors.  If I do the following aws cli command, I get a list of items in the s3 bucket:
`aws s3 ls s3://mybucket --profile abc`

If I do this in rocfl, 
`rocfl -p abc -b mybucket -R us-east-1 ls`

then the following error occurs:
```
[ERROR] Failed to list storage root extensions: Request ID: None Body: <?xml version="1.0" encoding="UTF-8"?>
    <Error><Code>AccessDenied</Code><Message>Access Denied</Message><RequestId>XXXX (omitted)</RequestId><HostId>ALONGDIGESTTHATIREMOVED</HostId></Error>
[ERROR] Failed to load OCFL layout: unknown variant `NNNN-flat-omit-prefix-storage-layout`, expected one of `0002-flat-direct-storage-layout`, `0004-hashed-n-tuple-storage-layout`, `0003-hash-and-id-n-tuple-storage-layout`, `0006-flat-omit-prefix-storage-layout` at line 2 column 54
```

--- a/src/cmd/mod.rs
+++ b/src/cmd/mod.rs
@@ -220,9 +220,9 @@ fn create_s3_repo(config: &Config) -> Result<OcflRepo> {
     OcflRepo::s3_repo(
         region,
         config.bucket.as_ref().unwrap(),
-        config.profile.as_deref(),
         config.root.as_deref(),
         config.staging_root.as_ref().unwrap(),
+        config.profile.as_deref(),
     )
 }
 
@@ -233,10 +233,10 @@ fn init_s3_repo(config: &Config, layout: Option<StorageLayout>) -> Result<OcflRe
     OcflRepo::init_s3_repo(
         region,
         config.bucket.as_ref().unwrap(),
-        config.profile.as_deref(),
         config.root.as_deref(),
         config.staging_root.as_ref().unwrap(),
         layout,
+        config.profile.as_deref(),
     )
 }
 

--- a/src/ocfl/repo.rs
+++ b/src/ocfl/repo.rs
@@ -94,14 +94,14 @@ impl OcflRepo {
     pub fn init_s3_repo(
         region: Region,
         bucket: &str,
-        profile: Option<&str>,
         prefix: Option<&str>,
         staging_root: impl AsRef<Path>,
         layout: Option<StorageLayout>,
+        profile: Option<&str>,
     ) -> Result<Self> {
         Ok(Self {
             staging_root: staging_root.as_ref().to_path_buf(),
-            store: Box::new(S3OcflStore::init(region, bucket, profile, prefix, layout)?),
+            store: Box::new(S3OcflStore::init(region, bucket, prefix, layout, profile)?),
             staging: OnceCell::default(),
             staging_lock_manager: OnceCell::default(),
             use_backslashes: false,
@@ -115,13 +115,13 @@ impl OcflRepo {
     pub fn s3_repo(
         region: Region,
         bucket: &str,
-        profile: Option<&str>,
         prefix: Option<&str>,
         staging_root: impl AsRef<Path>,
+        profile: Option<&str>,
     ) -> Result<Self> {
         Ok(Self {
             staging_root: staging_root.as_ref().to_path_buf(),
-            store: Box::new(S3OcflStore::new(region, bucket, profile, prefix)?),
+            store: Box::new(S3OcflStore::new(region, bucket, prefix, profile)?),
             staging: OnceCell::default(),
             staging_lock_manager: OnceCell::default(),
             use_backslashes: false,

--- a/src/ocfl/store/s3.rs
+++ b/src/ocfl/store/s3.rs
@@ -58,10 +58,10 @@ impl S3OcflStore {
     pub fn new(
         region: Region,
         bucket: &str,
-        profile: Option<&str>,
         prefix: Option<&str>,
+        profile: Option<&str>,
     ) -> Result<Self> {
-        let s3_client = S3Client::new(region, bucket, profile, prefix)?;
+        let s3_client = S3Client::new(region, bucket, prefix, profile)?;
 
         check_extensions(&s3_client);
         let storage_layout = load_storage_layout(&s3_client);
@@ -79,11 +79,11 @@ impl S3OcflStore {
     pub fn init(
         region: Region,
         bucket: &str,
-        profile: Option<&str>,
         prefix: Option<&str>,
         layout: Option<StorageLayout>,
+        profile: Option<&str>,
     ) -> Result<Self> {
-        let s3_client = S3Client::new(region, bucket, profile, prefix)?;
+        let s3_client = S3Client::new(region, bucket, prefix, profile)?;
 
         init_new_repo(&s3_client, layout.as_ref())?;
 
@@ -550,8 +550,8 @@ impl S3Client {
     fn new(
         region: Region,
         bucket: &str,
-        profile: Option<&str>,
         prefix: Option<&str>,
+        profile: Option<&str>,
     ) -> Result<Self> {
         Ok(S3Client {
             s3_client: create_rusoto_client(region, profile),

--- a/src/ocfl/store/s3.rs
+++ b/src/ocfl/store/s3.rs
@@ -595,18 +595,23 @@ impl S3Client {
                         ..Default::default()
                     }))?;
 
+            let prefix_offset = if self.prefix.is_empty() {
+                0
+            } else {
+                self.prefix.len() + 1
+            };
+
             if let Some(contents) = &result.contents {
                 for object in contents {
-                    objects.push(object.key.as_ref().unwrap()[self.prefix.len() + 1..].to_owned());
+                    objects.push(object.key.as_ref().unwrap()[prefix_offset..].to_owned());
                 }
             }
 
             if let Some(prefixes) = &result.common_prefixes {
                 for prefix in prefixes {
                     let length = prefix.prefix.as_ref().unwrap().len() - 1;
-                    directories.push(
-                        prefix.prefix.as_ref().unwrap()[self.prefix.len() + 1..length].to_owned(),
-                    );
+                    directories
+                        .push(prefix.prefix.as_ref().unwrap()[prefix_offset..length].to_owned());
                 }
             }
 

--- a/tests/s3-tests.rs
+++ b/tests/s3-tests.rs
@@ -602,7 +602,7 @@ fn default_repo(prefix: &str, staging: impl AsRef<Path>) -> OcflRepo {
 }
 
 fn init_repo(prefix: &str, staging: impl AsRef<Path>, layout: Option<StorageLayout>) -> OcflRepo {
-    OcflRepo::init_s3_repo(REGION, &bucket(), None, Some(prefix), staging, layout).unwrap()
+    OcflRepo::init_s3_repo(REGION, &bucket(), Some(prefix), staging, layout, None).unwrap()
 }
 
 fn s3_prefix() -> String {

